### PR TITLE
fix(clerk-js): Align some sections in User Profile with the designs

### PR DESCRIPTION
**Pull request by octoper**
## Description

This PR adds a hover state for all profile section items that include an action menu (eg. Emails, Phone numbers etc.)

## Before
CleanShot 2024-02-16 at 15 51 09@2x

## After
CleanShot 2024-02-16 at 15 53 57@2x

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/clerk-js/src/ui/components/UserProfile/ConnectedAccountsSection.tsx [hunk 1]
```diff
@@ -61,7 +61,13 @@ export const ConnectedAccountsSection = withCardStateProvider(() => {
 
             return (
               <Action.Root key={account.id}>
-                <ProfileSection.Item id='connectedAccounts'>
+                <ProfileSection.Item
+                  id='connectedAccounts'
+                  sx={t => ({
+                    borderRadius: t.radii.$lg,
+                    ':hover': { backgroundColor: t.colors.$neutralAlpha50 },
```
Comment (desiprisg) [inline packages/clerk-js/src/ui/components/UserProfile/ConnectedAccountsSection.tsx:68]: Can we avoid the duplication here by putting the styles inside `ProfileSection.Item` ?

File: packages/clerk-js/src/ui/elements/ArrowBlockButton.tsx [hunk 2]
```diff
@@ -62,6 +62,7 @@ export const ArrowBlockButton = React.forwardRef<HTMLButtonElement, ArrowBlockBu
           justifyContent: 'center',
           borderColor: theme.colors.$neutralAlpha100,
           alignItems: 'center',
+          padding: `${theme.space.$1x5} ${theme.space.$3} ${theme.space.$1x5} ${theme.space.$2x5}`,
```
Comment (desiprisg) [inline packages/clerk-js/src/ui/elements/ArrowBlockButton.tsx:65]: Why is this necessary? Isn't it going to make most screens deviate from designs now?

File: packages/clerk-js/src/ui/elements/Section.tsx [hunk 3]
```diff
@@ -128,7 +128,7 @@ const ProfileSectionItem = (props: ProfileSectionItemProps) => {
           justifyContent: 'space-between',
           width: '100%',
           alignItems: 'center',
-          padding: `${t.space.$1x5} ${t.space.$none} ${t.space.$1x5} ${t.space.$4}`,
+          padding: `${t.space.$1x5} ${t.space.$3} ${t.space.$1x5} ${t.space.$2x5}`,
```
Comment (desiprisg) [inline packages/clerk-js/src/ui/elements/Section.tsx:131]: Ditto

File: packages/clerk-js/src/ui/elements/ArrowBlockButton.tsx (hunk 2 above)
Comment (octoper) [reply packages/clerk-js/src/ui/elements/ArrowBlockButton.tsx:65]: This is the padding that the designs have.

File: packages/clerk-js/src/ui/elements/Section.tsx (hunk 3 above)
Comment (octoper) [reply packages/clerk-js/src/ui/elements/Section.tsx:131]: This change is made to be completely aligned with the designs

File: packages/clerk-js/src/ui/components/UserProfile/ConnectedAccountsSection.tsx (hunk 1 above)
Comment (octoper) [reply packages/clerk-js/src/ui/components/UserProfile/ConnectedAccountsSection.tsx:68]: We can't move them in `ProfileSection.Item` as we use it for stuff that don't need the hover state and the border radius but we can create a new component or add a prop to add those styles.

Comment (desiprisg) [reply packages/clerk-js/src/ui/components/UserProfile/ConnectedAccountsSection.tsx:68]: We can move them in and add a prop or make a small wrapper as you said. I prefer the wrapper tbh.

**Comment by desiprisg** [comment]
!preview

**Comment by clerk-cookie** [comment]
Hey @desiprisg, your preview is currently deploying and should be available shortly.

| Status | Preview | Updated (UTC) |
| :-- | :-- | :-- |
| :cook: _Deploying..._ | [Inspect](https://javascript-preview-no3wkrntq.clerkstage.dev) | Feb 16, 2024 04:12 PM |

File: packages/clerk-js/src/ui/components/UserProfile/ConnectedAccountsSection.tsx (hunk 1 above)
Comment (octoper) [reply packages/clerk-js/src/ui/components/UserProfile/ConnectedAccountsSection.tsx:68]: The only thing I don't like about the wrapper is that it's can be a bit confusing as it would be the same thing, the hover state is like having a variant.

**Comment by anagstef** [comment]
!preview

**Comment by octoper** [comment]
!preview

**Comment by anagstef** [comment]
!preview

**Review by anagstef — APPROVED** [review]
